katacomb/katacomb/qa_report.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import os
+import shlex
+import sys
+
+import contextlib
+
+import Radio_continuum_validation as rcv
+import katsdpimageutils.primary_beam_correction as pbc
+from katacomb.aips_export import _make_pngs, FITS_EXT, PNG_EXT
+
+log = logging.getLogger('katacomb')
+
+
+@contextlib.contextmanager
+def log_qa(logger):
+    """Trap QA stdout and stderr messages and send them to logger."""
+    original = sys.stdout
+    original_err = sys.stderr
+    sl = StreamToLogger(logger, logging.INFO)
+    sys.stdout = sl
+
+    sl = StreamToLogger(logger, logging.ERROR)
+    sys.stderr = sl
+
+    yield
+    sys.stdout = original
+    sys.stderr = original_err
+
+
+def make_pbeam_images(metadata, in_dir, out_dir):
+    """Write primary beam corrected images.
+
+    Make a single plane fits image, a png and a thumbnail
+    Parameters
+    ----------
+    metadata : dict
+        dictionary containing pipeline metadata
+    in_dir : str
+        path containing pipeline output files
+    out_dir : str
+        path to write primary beam corrected images to
+    """
+    filenames = metadata['FITSImageFilename']
+    targets = metadata['Targets']
+    for target, out_file in zip(targets, filenames):
+        out_filebase = os.path.splitext(out_file)[0]
+        log.info('Write primary beam corrected FITS output: %s',
+                 out_filebase + '_PB' + FITS_EXT)
+
+        in_path = os.path.join(in_dir, out_file)
+        pbc_path = os.path.join(out_dir, out_filebase + '_PB' + FITS_EXT)
+        bp, raw_image = pbc.beam_pattern(in_path)
+        pbc_image = pbc.primary_beam_correction(bp, raw_image, px_cut=0.1)
+        pbc.write_new_fits(pbc_image, in_path, outputFilename=pbc_path)
+        log.info('Write primary beam corrected PNG output: %s',
+                 out_filebase + '_PB' + PNG_EXT)
+        caption = f'{target}'
```

Comment:
Can we keep the captions in the same format as the uncorrected images:
something like:
`caption = f'{target} Continuum ({center_freq} MHz) (PB Corrected)'`

You can get center frequency from the FITS header of the image.

Reply:
Done